Lib/typing.py

```diff
@@ -2252,10 +2252,12 @@ def __new__(cls, name, bases, ns, total=True):
         Subclasses and instances of TypedDict return actual dictionaries.
         """
         for base in bases:
-            if type(base) is not _TypedDictMeta:
+            if not (type(base) is _TypedDictMeta or base is Generic):
                 raise TypeError('cannot inherit from both a TypedDict type '
                                 'and a non-TypedDict base class')
-        tp_dict = type.__new__(_TypedDictMeta, name, (dict,), ns)
+        
+        generic_base = (Generic,) if _collect_type_vars(ns.get("__orig_bases__", ())) else ()
```

Comment:
I have found an bug with this if statement:

When we have concrete generic that does not have any typevars it will not inherit from `Generic`, so it's lead to issue:
```py
from typing import TypedDict, Generic, TypeVar

T = TypeVar("T")


class BaseGeneric(TypedDict, Generic[T]):
    a: T
    b: int


class Delivered(BaseGeneric[int]):
    c: float

Delivered[int]  # will not fail and will call dict.__class_getitem__
```
But expected behaviour is:
```py
Delivered[int]  # TypeError: <class '__main__.Delivered'> is not a generic class
```

It can be fixed with following condition:
```suggestion
        if '__orig_bases__' in ns and any(issubclass(b, Generic) for b in bases):
            generic_base = (Generic,)
        else:
            generic_base = ()
```

Reply:
Good catch! Thank you! But I'm wondering if we are conflating two issues now because original TypedDict should not have been able to subscript from the beginning.

Reply:
Yup, totally agree with you. Possible solution I have described here - https://github.com/python/cpython/pull/27663#issuecomment-913627614